lib/pbench/test/unit/agent/task/test_results_move.py

```diff
@@ -250,3 +267,128 @@ def test_results_move(monkeypatch, caplog, setup):
+            responses.PUT,
+            f"{TestResultsMove.SRVR_TEXT}/api/v1/upload/{script}_{config}_{date}.tar.xz",
+            status=200,
+        )
+
+        runner = CliRunner(mix_stderr=False)
+
+        # Test --no-delete
+        result = runner.invoke(
+            main,
+            args=[
+                TestResultsMove.CTRL_SWITCH,
+                TestResultsMove.CTRL_TEXT,
+                TestResultsMove.TOKN_SWITCH,
+                TestResultsMove.TOKN_TEXT,
+                TestResultsMove.DELN_SWITCH,
+                TestResultsMove.SRVR_SWITCH,
+                TestResultsMove.SRVR_TEXT,
+            ],
+        )
+        assert (
+            result.exit_code == 0
+        ), f"Expected a successful operation, exit_code = {result.exit_code:d}, stderr: {result.stderr}, stdout: {result.stdout}"
+        assert (
+            result.stdout
+            == "Status: total # of result directories considered 1, successfully copied 1, encountered 0 failures\n"
+        )
+        # This should raise an unexpected exception if it was not created.
+        (pbrun / f"{name}.copied").unlink()
+
+    @staticmethod
+    @responses.activate
+    def test_results_move_relay(monkeypatch, caplog, setup):
+        monkeypatch.setenv("_pbench_full_hostname", "localhost")
+        monkeypatch.setattr(datetime, "datetime", MockDatetime)
+
+        # In order for a pbench tar ball to be moved/copied to a pbench-server
+        # the run directory has to have one file in it, a "metadata.log" file.
+        # We make a run directory and populate it with our test specific
+        # information.
+        pbrun = setup["tmp"] / "var" / "lib" / "pbench-agent"
+        script = "pbench-user-benchmark"
+        config = "test-results-move"
+        date = "YYYY.MM.DDTHH.MM.SS"
+        name = f"{script}_{config}_{date}"
+        res_dir = pbrun / name
+        res_dir.mkdir(parents=True, exist_ok=True)
+        mlog = res_dir / "metadata.log"
+        mlog.write_text(mdlog_tmpl.format(**locals()))
+
+        caplog.set_level(logging.DEBUG)
+
+        responses.add(
+            responses.PUT, re.compile(f"{TestResultsMove.RELAY_TEXT}/[a-z0-9]+")
+        )
+
+        runner = CliRunner(mix_stderr=False)
+
+        # Test --no-delete
+        result = runner.invoke(
+            main,
+            args=[
+                TestResultsMove.CTRL_SWITCH,
+                TestResultsMove.CTRL_TEXT,
+                TestResultsMove.DELN_SWITCH,
+                TestResultsMove.RELAY_SWITCH,
+                TestResultsMove.RELAY_TEXT,
+            ],
+        )
+
+        # We expect two PUT calls using the relay base URI: a JSON manifest
+        # object followed by the tarball. Responses stores "calls" in reverse
+        # order. While we can effect random access by searching for URIs, we
+        # can't easily know either SHA256 as the tarball and manifest are both
+        # generated internally. Instead, we traverse the list of calls in
```

Comment:
The reason why we cannot know is that the hash value is calculated from the file contents which includes timestamps, in the case of the tar file, so the value is different every time the test is run?

Reply:
Yes; if we were using pre-constructed tarballs we could "fake it", but I actually like this better.